src/Generalize.js

```diff
@@ -44,11 +44,49 @@ L.MarkerGeneralizeGroup = L.FeatureGroup.extend({
 
         this._zoomReady = {};
 
+        this.setMaxZoom(options.maxZoom);
+        this.setMinZoom(options.minZoom);
+
         this.on('invalidationFinish', function() {
             this._map.getPanes().markerPane.style.display = 'block';
         });
     },
 
+    setMaxZoom: function(zoom) {
+        if (isNaN(zoom) || zoom > 19) {
+            this._maxZoom = 19;
+            this._userMaxZoom = false;
+        } else {
+            this._maxZoom = zoom;
+            this._userMaxZoom = true;
+        }
```

Comment:
By the way, what happens if we don't pass zooms in options at all? We will assign `_userMaxZoom` to `true` instead of just ignoring it.